students/Net_Michael/lesson4/mailroom2.py

```diff
@@ -1,132 +1,147 @@
+
+def quit_sel():
+    print("Exit")
+
+###############################################
+### following modules are supplementary embded on the above functions
+###############################################
 
+def letter_format( donor, gift):
+    print('''
+          Dear {},
+          Thank you for your very kind donation of ${:,.2f}.
+          It will be put to very good use.
+                                Sincerely,
+                                  -The Team'''.format(donor,gift))
 
-# In[ ]:
+# reads text file from the file path
+def read_donor_info( donation_directory):
+    donor_lists = open(donation_directory)
+    donations = donor_lists.read()
+    donor_lists.close()
+    return(donations)
 
+# extract the donation information of one donor
+def one_donor_info( donor_folder, file_name):
+    # donor_folder: name of the folder in the directory holding donor files
+    #file_name: the file name of the donor (full name of the donor,
+    #              "firstname_last_name.txt")
+    donation_directory = os.path.join( os.getcwd(), donor_folder, file_name)
+    donation_filename = os.path.basename( file_name)
+    donor_name = os.path.splitext( donation_filename)[0].split("_")
+    donor_collect = read_donor_info( donation_directory)
+    donations = donor_collect.split(", ")
+    donation_out = [eval(item) for item in donations]
+    donor_info = {'donor': donor_name, 'gifts': donation_out}
+    return(donor_info)
 
+# creates a loop to get the donation ifnormation of
+# all donors in the directory
 
+def all_donors_info( donor_folder):
+    # donor_folder: name of the folder in the directory holding donor files
+    donation_dir = os.path.join( os.getcwd(), donor_folder)
+    donor_files = os.listdir( donation_dir)
+    donors_gifts = defaultdict( list)
+    for file_name in donor_files:
+        donor_gift = one_donor_info( donor_folder, file_name)
+        donor_name = " ".join( donor_gift["donor"])
+        donation = donor_gift["gifts"]
+        donors_gifts[donor_name] = donation
+    return(donors_gifts)
 
+# switch function will select the fumctions above
+# depending on the selection of user
+
+switch_func_dict = {
+        1: create_report,
+        2: thank_all_letter,
+        3: thank_one_letter,
+        4: quit_sel,
+    }
+
+def output_entry_value():
+    print('''Please choose one of the following numbers:
+             For report select - 1,
+             Send thank you letter to all donors - 2,
+             Send thank you letter to selected donor - 3,
+             Quit - 4
+
+             ''')
+    sel_num = input("Enter one of the four numbers: ")
+    sel_num = eval(sel_num)
+    return(sel_num)
+
+
+# any of the follwing for choices can be executed for desired output
+
+if __name__ == "__main__":
+    sel_choice = output_entry_value()
+    if sel_choice <= 4:
+        switch_func_dict.get(sel_choice)()
+    else:
+        print("Select an integer between 1 and 4")
```

Comment:
This is looking good, but you'll want to be able to add donors and donations as well as produce the report and letters.